### PR TITLE
Manila: add extend and shrink share functionality

### PR DESCRIPTION
**Pull request by gman0**
For #1174 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

Extend:
* example request: https://github.com/openstack/manila/blob/master/api-ref/source/samples/share-actions-extend-request.json
* API v2 `extend`: https://github.com/openstack/manila/blob/master/manila/api/v2/shares.py#L385
* `extend` implementation: https://github.com/openstack/manila/blob/master/manila/share/api.py#L1765

Shrink:
* example request: https://github.com/openstack/manila/blob/master/api-ref/source/samples/share-actions-shrink-request.json
* API v2 `shrink`: https://github.com/openstack/manila/blob/master/manila/api/v2/shares.py#L397
* `shrink` implementation: https://github.com/openstack/manila/blob/master/manila/share/api.py#L1823

/cc @jtopjian 

Ready for review

### CODE REVIEW

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 1]
```diff
@@ -267,3 +267,77 @@ func ListAccessRights(client *gophercloud.ServiceClient, id string) (r ListAcces
 	})
 	return
 }
+
+// ExtendOptsBuilder allows extensions to add additional parameters to the
+// Extend request.
+type ExtendOptsBuilder interface {
+	ToExtendMap() (map[string]interface{}, error)
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: `ToShareExtendMap`

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 2]
```diff
@@ -267,3 +267,77 @@ func ListAccessRights(client *gophercloud.ServiceClient, id string) (r ListAcces
...
+}
+
+// ExtendOpts contains the options for creation of a Extend request.
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: How about:

ExtendOpts contains options for extending a Share.

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 3]
```diff
@@ -267,3 +267,77 @@ func ListAccessRights(client *gophercloud.ServiceClient, id string) (r ListAcces
...
+		OkCodes: []int{202},
+	})
+
+	return
+}
+
+// ShrinkOptsBuilder allows extensions to add additional parameters to the
+// Shrink request.
+type ShrinkOptsBuilder interface {
+	ToShrinkMap() (map[string]interface{}, error)
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: `ToShareShrinkMap`

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 4]
```diff
@@ -267,3 +267,77 @@ func ListAccessRights(client *gophercloud.ServiceClient, id string) (r ListAcces
...
+}
+
+// ShrinkOpts contains the options for creation of a Shrink request.
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: How about:

ShrinkOpts contains options for shrinking a Share

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 1 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/requests.go]: done

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 2 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/requests.go]: done

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 3 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/requests.go]: done

File: openstack/sharedfilesystems/v2/shares/requests.go (hunk 4 above)
Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/requests.go]: done

**Comment by gman0** [comment]
@jtopjian Thanks for the review! PTAL

> To confirm: you've tested this implementation?

Yes, I've tested both extend and shrink operations in our on-site OpenStack cloud.

**Comment by gman0** [comment]
Thanks!

**Pull request merged**